### PR TITLE
Schema

### PR DESCRIPTION
**Pull request by AdnanOlarmmi**
Learning objectives
Install Ruby on Rails framework.
Use Ruby Gems as a software packages system.
Understand what ORM is.
Use database migration files to maintain database schema.

-Workflow

Project requirements
Be sure to reference the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) from the sneak peek if you have any questions about table or column names or data relationships.

Create a new app. Give it a meaningful name.
Initialize your project with Git.
Make sure that your project has Postgres database set up.
Build your project schema.
Create and run the necessary migration files.
Table and column names should match the ERD diagram.
note: photo for users table should be a link to an image
Foreign keys should be included.
All columns that are foreign keys should have a corresponding index.

### CODE REVIEW

**Review by OLIPLICHE — COMMENTED** [review]
Hi @AdnanOlarmmi , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

**You have submitted a project, that does not contain the ReadMe template provided by Microverse**

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: db/migrate/20230118195424_add_user_ref_likes.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class AddUserRefLikes < ActiveRecord::Migration[7.0]
+  def change
+  end
+end
```
Comment (andersonlebon) [inline db/migrate/20230118195424_add_user_ref_likes.rb]: - [ ] Kindly reference the user in the likes table, currently this migration is empty, also make sure to use the keyword(auther_id) to make the reference. hint :point_down: 

```rb

class AddUserRefLikes < ActiveRecord::Migration[7.0]
  def change
    add_reference :likes, :author, null: false, foreign_key: { to_table: 'users' }
  end
end
```

File: db/migrate/20230118195359_add_user_ref_to_posts.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
```
Comment (andersonlebon) [inline db/migrate/20230118195359_add_user_ref_to_posts.rb]: - [ ] Nice job :clap: on referencing the user table to its posts, however, the reference should be insured by the keyword `author_id` as in the [EDR Diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). hint to solve it :point_down: 

```rb
class AddUserRefToPosts < ActiveRecord::Migration[7.0]
  def change
    add_reference :posts, :author, null: false, foreign_key: { to_table: 'users' }
  end
end
```

File: db/migrate/20230118195231_add_user_ref_to_comments.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToComments < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :user, null: false, foreign_key: true
```
Comment (andersonlebon) [inline db/migrate/20230118195231_add_user_ref_to_comments.rb]: - [ ] Nice job :clap: on referencing the user table to its comments, however, the reference should be insured by the keyword `author_id` as in the [EDR Diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). hint to solve it :point_down: 

```rb
class AddUserRefToComments < ActiveRecord::Migration[7.0]
  def change
    add_reference :comments, :author, null: false, foreign_key: { to_table: 'users' }
  end
end
```

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @AdnanOlarmmi 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230118195359_add_user_ref_to_posts.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: {to_table: 'users'}
```
Comment (codepantha) [inline db/migrate/20230118195359_add_user_ref_to_posts.rb]: - [ ] Like the previous reviewer suggested, please ensure that the reference is implemented with the `author_id` as shown in the [ERD Diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png)

File: db/migrate/20230118195231_add_user_ref_to_comments.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToComments < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :user, null: false, foreign_key: {to_table: 'users'}
```
Comment (codepantha) [inline db/migrate/20230118195231_add_user_ref_to_comments.rb]: - [ ] Nice job trying to implement the suggested change but you missed the part where the reference should be implemented with the `author_id` as shown in the [ERD Diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png)

File: db/migrate/20230118195424_add_user_ref_likes.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefLikes < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :user, null: false, foreign_key: {to_table: 'users'}
```
Comment (codepantha) [inline db/migrate/20230118195424_add_user_ref_likes.rb]: - [ ] Like the previous reviewer suggested, please ensure that the reference is implemented with the `author_id` as shown in the [ERD Diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png)

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @AdnanOlarmmi 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors
✔️ Professional README

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @AdnanOlarmmi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**